core/include/traccc/fitting/kalman_filter/gain_matrix_updater.hpp

```diff
@@ -37,20 +38,24 @@ struct gain_matrix_updater {
     /// @param bound_params bound parameter
     ///
     /// @return true if the update succeeds
+    template <typename track_state_backend_t>
```

Comment:
What is this `track_state_backend_t` here?

Reply:
It is the `BASE` type from `track_state_collection.hpp`. "Backend" seems like an appropriate term for it.

The templating is needed here because in the CKF this code gets called on a "standalone" `traccc::edm::track_state` object. While in the KF it is called on an element of an SoA container. The "backend" describes that in one case the `trk_state` is operating on values, and in the other it operates on references.

Reply:
We had this exact discussion on a previous PR already.

If not for this type of templating, then we'd need a concept for `traccc::edm::track_state<T>`. But the intention here is really that the user should only be able to provide `traccc::edm::track_state` objects here. Not something that maybe has the same accessor functions as it does.